lib/libutee/tee_api_operations.c

```diff
@@ -377,14 +394,20 @@ TEE_Result TEE_SetOperationKey2(TEE_OperationHandle operation,
 		TEE_ObjectInfo key_info1;
 		TEE_ObjectInfo key_info2;
 
-		TEE_GetObjectInfo(key1, &key_info1);
+		res = TEE_GetObjectInfo1(key1, &key_info1);
+		if (res != TEE_SUCCESS)
+			goto err;
+
 		/* Supplied key has to meet required usage */
 		if ((key_info1.objectUsage & operation->info.
 		     requiredKeyUsage) != operation->info.requiredKeyUsage) {
 			TEE_Panic(0);
 		}
 
-		TEE_GetObjectInfo(key2, &key_info2);
+		res = TEE_GetObjectInfo1(key2, &key_info2);
```

Comment:
If this function call returns TEE_ERROR_CORRUPT_OBJECT, we should return TEE_ERROR_CORRUPT_OBJECT_2 instead?